lms/djangoapps/grades/models.py

```diff
@@ -121,6 +123,7 @@ def create_from_blockrecords(self, blocks):
         """
         model, _ = self.get_or_create(
             hashed=blocks.hash_value,
+            version=VISIBLE_BLOCKS_VERSION,
```

Comment:
As discussed, we need to determine what the uniqueness constraint is on the model.  If it remains as just `hashed`, then (1) `version` should be included in `defaults` and (2) version should be included in the hash value.  If the uniqueness constraint is the `(hashed, version)` tuple, then there may be issues with the foreign key constraint on the related model.